crypto/s2n_libcrypto.c

```diff
@@ -153,6 +153,20 @@ S2N_RESULT s2n_libcrypto_init(void)
     return S2N_RESULT_OK;
 }
 
+#if S2N_OPENSSL_VERSION_AT_LEAST(3, 0, 0)
+int s2n_libcrypto_cleanup_cb(OSSL_PROVIDER *provider, void *cbdata) {
+    return OSSL_PROVIDER_unload(provider);
+}
+#endif
+
+S2N_RESULT s2n_libcrypto_cleanup(void) {
+#if S2N_OPENSSL_VERSION_AT_LEAST(3, 0, 0)
+    RESULT_GUARD_OSSL(OSSL_PROVIDER_do_all(NULL, *s2n_libcrypto_cleanup_cb, NULL), S2N_ERR_ATEXIT);
```

Comment:
"error calling atexit" doesn't seem like the right error here. That assumes that this method is called by atexit. It is called by atexit, but that happens elsewhere in the codebase :)

Reply:
Should i be creating a new error here? Wasnt sure if that was overkill

Reply:
Maybe you could modify S2N_ERR_OSSL_LOAD_PROVIDER to apply to both load and unload. Like maybe just "S2N_ERR_OSSL_PROVIDER"

Reply:
modify S2N_ERR_OSSL_LOAD_PROVIDER.. I like that